src/Services/Azure/Authentication/CustomChainedCredential.cs

```diff
@@ -18,22 +19,72 @@ namespace AzureMcp.Services.Azure.Authentication;
 /// </remarks>
 public class CustomChainedCredential(string? tenantId = null) : TokenCredential
 {
-    private readonly ChainedTokenCredential _chainedCredential = CreateChainedCredential(tenantId);
+    public override AccessToken GetToken(TokenRequestContext requestContext, CancellationToken cancellationToken)
+    {
+        ChainedTokenCredential chainedCredential = CreateChainedCredential(tenantId);
```

Comment:
Do we need to create the credential instance each time GetToken is called, or is it possible to cache this?

Reply:
Do we need to worry about concurrent access to the _chainedCredential instance? Or is azmcp implemented to be essentially single threaded for now?

Reply:
The chained token credential is using impl from Identity SDK which is thread safe by design.

Reply:
I updated the code in this PR to cache the credential object. Let's add synchronization in a follow up PR.